contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+    the upgrade to cluster. Also the logs of the upgrade are saved
+    on the pod. 
+
+    Note: This replaces the script based upgrades from OpenEBS 1.0. 
+    Sample Kubernetes YAMLs for upgrading various resources can be
+    found [here](../../../../k8s/upgrades/1.0.0-1.1.0/).  
+
+    Status: Available in 1.1 and supports upgrading of Jiva Volumes,
+    cStor Volumes and cStor pools from 1.0 to 1.1
+
+3.  A custom resource called `UpgradeTask` will be defined with the 
+    details of the object being upgraded, and will in turn be updated
+    with the status of the upgrade by the upgrade-container. This 
+    will allow for tools to be written to capture the status of the
+    upgrades and take any correction actions. One of the consumer
+    for this UpgradeTask is openebs-operator itself that will automate
+    the upgrade of all resources. 
+
+    The `UpgradeTask` resource will be created by the Upgrade Job. 
+
+    Status: Under Development, planned for 1.2
+
+4.  Improvements to the backward compatibility checks added to the 
+    OpenEBS Control Plane in (1). The backward compatibility checks
+    will involve checking for multiple resources and this process
+    is triggered on a restart. This process will be optimized to 
+    use a flag to check if the backward compatibility checks are 
+    indeed required. On the resource being managed, the following 
+    internal spec will be maintained that indicates if the backward
+    compatibility checks need to be maintained. 
+
+    ```
+    versionDetails:
+      #Indicates if the resource should be auto upgraded 
+      #More on this below (6). Default is false.
+      #autoUpgrade:
+      #Indicates the current version of the resource.
+      currentVersion: 
+      #Indicates the running version of the controller/component
+      #as part of the start-up, this flag will be changed
+      #running version if autoUpgrade is enabled. 
+      desiredVersion: 
+      #In some cases there can be a bunch of child resources
+      #this flag will be set to "no" when current != desired. 
+      #after upgrading all the child objects, this will be changed to yes. 
+      dependentsUpgraded:
+    ```
+  
+    Status: Under Development, planned for 1.2
+
+5.  Downgrading a version. There are scenarios where the volumes will
+    have to be downgraded to earlier versions. Some of the challenges
+    around this are after upgrading a resource with a breaking change, 
+    falling back to older version might make the resource un-readable. 
+    To avoid this, the earlier version of the resource will be saved
+    under a versioned name. When downgrading from higher (currentVersion)
+    to lower (desiredVersion), the backup copy of the resource will
+    be applied. 
+
+    Status: Under Development, planned for TBD
+
+6.  Automated upgrades of data plane components. OpenEBS operator 
+    will check for all the volumes and resources that need to be upgraded
```

Comment:
Are we finalizing this approach? Or we leave it to higher order components i.e. openebs operator

Reply:
This is a TBD.. we can raise another OEP or update with specifics of the implementation.